scripts/map_preprocess/geo_utils.R

```diff
@@ -76,12 +76,21 @@ points_to_svg_coords <- function(geojson_filepath, tsv_filepath){
 }
 
 geo2svg <- function(geojson_filepath, svg_filepath, w, h){
-  #tempjson <- file.path(tempdir(), 'temp.json')
-  #system(sprintf("ndjson-split 'd.features' \
-  #  < %s \
-  #  > %s", geojson_filepath, tempjson))
   
-  system(sprintf('geo2svg -w %s -h %s < %s > %s', w, h, geojson_filepath, svg_filepath))
+  svg_precision <- 2
+  tempjson <- tempfile(fileext = '.json')
+  tempsimple <- tempfile(fileext = 'simple.json')
+  tempquant <- tempfile(fileext = 'quant.json')
+  tempndjson <- tempfile(fileext = '.ndjson')
+  system(sprintf("geo2topo %s -o %s", geojson_filepath, tempjson))
+  system(sprintf("toposimplify -s 1e-4 -f  %s -o %s", tempjson, tempsimple))
```

Comment:
This is cool - would like to see a version of this make it into `vizlab` once we've given it some exercise. Could/should the `toposimplify -s XX` and `topoquantize XX` arguments be arguments to the `geo2svg()` R function? and something for `state_boundaries_scaled.json` and `svg_precision` too? after that, seems like it would be a pretty darn general and handy function.

Reply:
Yes, those args should definitely be exposed. I had a heck of a time trying to figure out how to get each state split up into a different `path` element (default in `geo2svg` is a single path that contains all of the features). It turns out the `-n` arg to `geo2svg` tells that function that you are passing it `.ndjson` instead of `geojson`. By the time I figured that out I didn't have time for cleanup, but yes, it should be done.

Reply:
:+1: That line-splitting thing would have taken me forever, too.